orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/aws/AmazonImageFinder.java

```diff
@@ -33,6 +33,9 @@
 
 @Component
 public class AmazonImageFinder implements ImageFinder {
+
+  private static final int MAX_SEARCH_RESULTS = 1000;
```

Comment:
Any thoughts on making this configurable? 1000 seems like a lot?

Reply:
I think in the beginning the 1000 limit was set in clouddriver for reading "enough" entries from cache before applying some filters in memory. Now that this proposal reads _all_ matching entries from cache, the limit is just useful for avoiding network congestion, so it can be decreased both here and in clouddriver.

Ideally, clouddriver's response should include a way for orca to know if the results were truncated or not, to avoid having them to agree in a common number. Here are a couple of alternatives without changing too much the existing api:

* Include in the response a [Link Header](https://tools.ietf.org/html/rfc5988#page-6). Link headers are a way for supplying pagination information instead of embedding that data in the response payload.
* Include in the response a custom header, like `X-SPINNAKER-TOTAL-COUNT` or `X-SPINNAKER-PARTIAL-RESULT`, useful for identifying if the given response is complete or not.

Reply:
With how Retrofit works, having response envelopes are far more preferential to using headers.

```
{
  "results": [ ... ]
  "totalCount": 12312312312312312312
}
```